src/components/Asset/RelatedAssets/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { Asset } from '@oceanprotocol/lib'
+import { generateBaseQuery, queryMetadata } from '@utils/aquarius'
+import { useUserPreferences } from '@context/UserPreferences'
+import { useAsset } from '@context/Asset'
+import styles from './index.module.css'
+import { useCancelToken } from '@hooks/useCancelToken'
+import AssetList from '@shared/AssetList'
+import { generateQuery } from './_utils'
+
+export default function RelatedAssets(): ReactElement {
+  const { asset } = useAsset()
+  const { chainIds } = useUserPreferences()
+  const newCancelToken = useCancelToken()
+
+  const [relatedAssets, setRelatedAssets] = useState<Asset[]>()
+  const [isLoading, setIsLoading] = useState<boolean>()
+
+  useEffect(() => {
+    if (!chainIds || !asset?.nftAddress || !asset?.nft || !asset?.metadata)
+      return
+
+    async function getAssets() {
+      setIsLoading(true)
+      const tagQuery = generateBaseQuery(
+        generateQuery(chainIds, asset.nftAddress, 4, asset.metadata.tags)
+      )
+      const tagResults = (await queryMetadata(tagQuery, newCancelToken()))
+        .results
+
+      if (tagResults.length === 4) {
```

Comment:
I tried to follow in the comments what the intention of this check is and found this (replace 3 with 4 as I increased it):

>  If we do the first query and we get 3 assets, then we don't do the second query.

But pretty sure this wasn't meant as "we get EXACTLY 3 assets". Cause that's pretty much never going to happen in reality. So the me sounds like this check here should rather be representation of "if there are 4 or more results"?

So:

```suggestion
      if (tagResults.length >= 4) {
```

Which then needs another tweak to limit the `tagResults` to the first 4, and then set it into state.

Reply:
The original think was the following: If the `size` is set to 4 in `generateQuery` then the maximum results the query should return is 4, although in many cases it will be less and then in that case we also do the owner query